### ISSUE TITLE
Update Lua to current 5.3.1

### ISSUE THREAD

**Issue by kkirsche** [Homebrew/homebrew#43712]
I don't think this should be that hard. I just don't know what the SHA values are for the OS specific portion of the formula to update it myself

**Comment by DomT4** [comment]
Lua53 currently resides in `Homebrew/versions` [here](https://github.com/Homebrew/homebrew-versions/blob/master/lua53.rb). Can be installed with `brew install homebrew/versions/lua53`.

**Comment by kkirsche** [comment]
Why does it default to a non-recent version then by default?

**Comment by DomT4** [comment]
There's a long backstory to this, but the core only *typically* supports one version of each formula.

In short: For a long time this one core formula was Lua's 5.1 branch as everything was compatible with it. Then I did a lot of work to introduce side-by-side Lua formulae so Lua could become a Lua51 formula and the Lua formula became Lua's 5.2 branch, however quite a few things refuse to build against Lua52 still and so we've had to keep a Lua51 formula in the core for longer than hoped.

Some time after that Lua released the 5.3 branch stable, but nothing in the core immediately needs it and quite a lot in the core wouldn't build against it at that point, so we filed it in `homebrew/versions`. When the ecosystem moves over from Lua 5.2 to 5.3 a bit more it is likely the Lua formula becomes Lua's 5.3 branch but we're not there yet.

Long-term, there will only ever be one version of Lua in the core. That this isn't already the case is just a historical relic from Lua's 5.1 branch being the default formula for a long time & certain things still requiring that.

The current Lua formula is on the 5.2 branch which is still supported and actively being updated; the last upstream release for the 5.2 branch was March this year.
